files/find.py

```diff
@@ -276,6 +285,12 @@ def main():
         else:
             module.fail_json(size=params['size'], msg="failed to process size")
 
+    patterns = params['patterns']
+    if patterns and patterns[0].startswith("~"):
+        patterns = re.compile(patterns[0][1:])
```

Comment:
im not sure if I'm reading this correctly, if the first pattern starts with ~ all patterns are considered regexes?